src/MessageEditor.js

```diff
@@ -14,23 +21,69 @@ class MessageEditor extends React.Component {
 
   componentDidUpdate(prevProps) {
     if (this.props.id !== prevProps.id && this.props.id) {
+      this.setState({
+        messages: [],
+        gauges: []
+      });
       api.fetch(`glyphs/${this.props.id}/messages`)
       .then(messages => this.setState({ messages: messages }));
 
       api.fetch(`glyphs/${this.props.id}/gauges`)
       .then(gauges => this.setState({ gauges: gauges }));
+
     }
   }
 
   render() {
-    const { messages, gauges } = this.state;
+    let { messages, gauges } = this.state;
+
+    const addVMessage = () => {
+      api.post(`glyphs/${this.props.id}/messages/`, {"pass" : this.inputRef.value, "text" : 'Default message', "probability" : 0}).then(result => {console.log(result)})
+      messages.push({"text" : 'Default message', "probability" : 0})
+      this.forceUpdate();
+    }
+
+    const password = () => {
+      this.forceUpdate();
+    }
 
     return (
       <div className="MessageEditor">
-        {messages.map(m => m.text)}
-
+        <form>
```

Comment:
I don't think we have much of a need of a `form` tag here, no?

Reply:
You're right, residual code again